app/src/test/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharactersPresenterTest.kt

```diff
@@ -15,65 +18,53 @@ import org.mockito.Mockito
 import org.mockito.Mockito.mock
 import org.mockito.Mockito.verify
 
-//TODO fix on second iteration
-// error: However, there was exactly 1 interaction with this mock:
-class CharacterPresenterTest {
 
+class CharactersPresenterTest {
 
-    private var view = mock(CharecterView::class.java)
+    companion object {
+        private const val EMPTY_MSG = ""
+    }
+    private var view = mock(CharactersView::class.java)
     private var characterServiceImp = mock(CharacterServicesImpl::class.java)
-    private lateinit var characterPresenter: CharacterPresenter
+    private lateinit var charactersPresenter: CharactersPresenter
     private lateinit var getCharacterServiceUseCase: GetCharacterServiceUseCase
 
-
     @Before
     fun setUp() {
-
         RxAndroidPlugins.setInitMainThreadSchedulerHandler { scheduler -> Schedulers.trampoline() }
-
         getCharacterServiceUseCase = GetCharacterServiceUseCase(characterServiceImp)
-        val subscriptions = mock(CompositeDisposable::class.java)
-        characterPresenter = CharacterPresenter(view, getCharacterServiceUseCase, subscriptions)
-
-
+        charactersPresenter = CharactersPresenter(view, CharactersModel(getCharacterServiceUseCase))
     }
 
     @Ignore
     fun reposeWithError() {
-        Mockito.`when`(getCharacterServiceUseCase.invoke()).thenReturn(Observable.error(Exception("")))
-        characterPresenter.init()
+        Mockito.`when`(getCharacterServiceUseCase.invoke()).thenReturn(Observable.error(Exception(EMPTY_MSG)))
```

Comment:
please don't use mockit use mockito kotlin2 from naarma

Reply:
In next PR's i will be working with unit test, and also will add mockito kotlin 2!